London10-Afsha-Hossain-JS-Core1-Coursework-Week3

Afsha10 opened the pull request.

Afsha10 committed:
Answered why the outputs in the exercises are undefined

Answered using comments why the outputs in the exercises are undefined

Afsha10 committed:
Completed exercise tasks B and C

Afsha10 committed:
Completed Exercises D and E and Mandatory task 1 and task 2

Afsha10 committed:
Completed the rest of the mandatory tasks

nsaimk commented on 2-mandatory/2-financial-times.js:
```diff
@@ -13,25 +20,72 @@ function potentialHeadlines(allArticleTitles) {
+//         fewestNumberOfWords = articleSplitter.length;
+
 function titleWithFewestWords(allArticleTitles) {
+    let shortestHeadline;
```
you could declare 'shortestHeadline' with const keyword instead let

moneyinthesky replied:
I think `shortestHeadline` needs to be declared with `let` here as we are re-assigning a new value to it a few lines later.

SalihaPopal reviewed (commented):
Well done!

SalihaPopal commented on 1-exercises/A-undefined/exercise.js:
```diff
@@ -30,7 +43,21 @@ function sayHelloToUser(user) {
+
+// The output in the example below contains undefined because letters is an array with three elements, indexed from 0 to 2. 
+// When we try to access the element at index 3 (arr[3]), which is beyond the last index of the letters array, JavaScript returns undefined because there is no value at that index.
+
```
Nice explanation.

SalihaPopal commented on 1-exercises/E-while-loop-with-array/exercise.js:
```diff
@@ -16,8 +16,14 @@ const BIRTHDAYS = [
+    }
 }
 
 console.log(findFirstJulyBDay(BIRTHDAYS)); // should output "July 11th"
```
Nice work, consice and good.

moneyinthesky commented on 2-mandatory/1-weather-report.js:
```diff
@@ -12,6 +12,12 @@
 */
 
```
Looks perfect 👍

moneyinthesky commented on 2-mandatory/2-financial-times.js:
```diff
@@ -5,6 +5,13 @@
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
```
This look good!
For extra practice, can you try re-writing this using the `filter` array method?

SalihaPopal reviewed (commented):
Great job!

moneyinthesky commented on 2-mandatory/2-financial-times.js:
```diff
@@ -13,25 +20,72 @@ function potentialHeadlines(allArticleTitles) {
 */
+//        ["The", "three", "questions", "that", "dominate", "investment"]
+//         fewestNumberOfWords = articleSplitter.length;
+
```
This is a very good attempt 👍  One small improvement:
- Could you try doing this without using `let fewestNumberOfWords = 1000;`
- You could either leave it uninitialised and check for `undefined` in your loop.. Or you could maybe initialise to the numberOfWords of the first title in the list.

moneyinthesky commented on 2-mandatory/2-financial-times.js:
```diff
@@ -13,25 +20,72 @@ function potentialHeadlines(allArticleTitles) {
+// 1. We are getting array of string. We need to find if the string includes numbers.
+// 2. Once we find a number, we will put the string to the new array.
+// 3. We are going to return the new array. New array is string with numbers.
+
```
Very nice 👍

moneyinthesky commented on 2-mandatory/2-financial-times.js:
```diff
@@ -13,25 +20,72 @@ function potentialHeadlines(allArticleTitles) {
+// 2. We need to find the number of Article titles.
+// 3. We divide (total character) by (number of article titles).
+// 4. Round it up
+
```
Looks very good!

moneyinthesky commented on 2-mandatory/3-stocks.js:
```diff
@@ -33,8 +33,24 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
+// 2. Divide the total price by the number of days i.e. 5
+
+
+
```
Very good variable names 😄 
One question: Do you need to re-calculate the average each time through the loop? Maybe it's enough to just calculate the average once at the end?

moneyinthesky commented on 2-mandatory/3-stocks.js:
```diff
@@ -48,7 +64,13 @@ function getAveragePrices(closingPricesForAllStocks) {
     The price change value should be rounded to 2 decimal places, and should be a number (not a string)
 */
```
Nice!
One question: Will this code - `closingPricesEachcompany[4] - closingPricesEachcompany[0]` - still work if we have more than 5 prices in each array? Or less than 5 prices?
Can you think of how to make sure this works for any numbers of prices?

moneyinthesky commented on 2-mandatory/3-stocks.js:
```diff
@@ -63,9 +85,35 @@ function getPriceChanges(closingPricesForAllStocks) {
+    //     }
+    //   }
+    //   arrayWithHighestValue.push(highestPrice);
+
```
This was a tough one!
Well done for completing it 👏

moneyinthesky commented:
Great job on this coursework 👏

SalihaPopal commented:
Thank you for your consideration and comments. I am going to apple what have you recommended me regarding some exercises.

Afsha10 committed:
Made some changes to exercises, mandatory and tried the extra part

Afsha10 committed:
Created a small test in weather report task

We checked if we have an array that is the same length as the number of cities passed in.
